Google.Api.Generator/Generation/ServiceSettingsCodeGenerator.cs

```diff
@@ -89,14 +96,40 @@ private IEnumerable<PropertyDeclarationSyntax> SettingsProperties()
             return _svc.Methods.SelectMany(PerMethod);
             IEnumerable<PropertyDeclarationSyntax> PerMethod(MethodDetails method)
             {
+                // Add the general per-method settings property.
                 var property = AutoProperty(Public, _ctx.Type<CallSettings>(), method.SettingsName, hasSetter: true);
                 // TODO: XmlDoc.
                 // TODO: Initialization.
                 yield return property;
-                // TODO: Extra properties for LRO and streaming methods.
+                // Add extra properties as required for special call types.
+                switch (method)
+                {
+                    // TODO: Extra properties for streaming methods.
+                    case MethodDetails.Lro lro:
+                        yield return LroSettingsProperty(lro);
+                        break;
+                }
             }
         }
 
+        private PropertyDeclarationSyntax LroSettingsProperty(MethodDetails.Lro method) =>
```

Comment:
Yoicks! (No ideas about how to simplify this, but eek!)

Reply:
Indeed. This is a downside of not using templates.